atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/atomic/CasCassImitatingConsensusForgettingStoreTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.atomic;
+
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.CheckAndSetException;
+import org.junit.Test;
+
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class CasCassImitatingConsensusForgettingStoreTest {
+    private static final Cell CELL = Cell.create(new byte[] {1}, new byte[] {0});
+    private static final byte[] VALUE = PtBytes.toBytes("VAL");
+    private static final byte[] VALUE_2 = PtBytes.toBytes("VAL2");
+    // solution to (1-x)^4 = 0.5
+    private static final double PROBABILITY_THROWING_ON_QUORUM_HALF = 0.16;
+    ConsensusForgettingStore neverThrowing = new CasCassImitatingConsensusForgettingStore(0.0);
+    CassandraImitatingConsensusForgettingStore sometimesThrowing =
+            new CasCassImitatingConsensusForgettingStore(PROBABILITY_THROWING_ON_QUORUM_HALF);
+
+    @Test
+    public void trivialGet() throws ExecutionException, InterruptedException {
+        assertThat(neverThrowing.get(CELL).get()).isEmpty();
+    }
+
+    @Test
+    public void canGetAfterMark() throws ExecutionException, InterruptedException {
+        neverThrowing.mark(CELL);
+        assertThat(neverThrowing.get(CELL).get()).contains(CasCassImitatingConsensusForgettingStore.IN_PROGRESS_MARKER);
+    }
```

Comment:
there is some duplication b/w this class and the pue class but would like to keep it this way for readability unless the reviewer feels too strongly about it.